src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+            " Bye. Hope to see you again soon!\n" +
+            HORIZONTAL_LINE;
+    private static final String HELP_MESSAGE = "Here's some tips on how to use me!\n\n" +
+            "todo [input]\n" +
+            "\t - Add a todo task to the list\n\n" +
+            "deadline [input] /by [input]\n" +
+            "\t - Add a deadline task by the given deadline\n\n" +
+            "event [input] /at [input]\n" +
+            "\t - Add an event task at the given time\n\n" +
+            "list\n" +
+            "\t - List out all the current tasks\n\n" +
+            "done [task_numbers]\n" +
+            "\t - Marks the given tasks as done\n\n" +
+            "bye\n" +
+            "\t - Terminates me :(\n";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // Initialize variables for program startup
+        System.out.println("Hello from\n" + LOGO + GREETING_MESSAGE);
+        boolean isProgramRunning = true;
+        String userInput;
+        List<Task> list = new ArrayList<>();
+        Scanner in = new Scanner(System.in);
+
+        while (isProgramRunning) {
+            userInput = in.nextLine().trim();
+            String userInputLowerCase = userInput.toLowerCase();
+
+            if (userInput.equalsIgnoreCase("bye")) {
+                System.out.println(GOODBYE_MESSAGE);
+                isProgramRunning = false;
+            } else if (userInput.equalsIgnoreCase("list")) {
+                printList(list);
+            } else if (userInputLowerCase.startsWith("done")) {
+                markTasksAsDone(userInput, list);
+            } else if (userInputLowerCase.startsWith("todo")) {
+                String taskToAdd = removeFirstWord(userInput);
+                addTask(TaskType.TODO, taskToAdd, list);
+            } else if (userInputLowerCase.startsWith("deadline")) {
+                String taskToAdd = removeFirstWord(userInput);
+                addTask(TaskType.DEADLINE, taskToAdd, list);
+            } else if (userInputLowerCase.startsWith("event")) {
+                String taskToAdd = removeFirstWord(userInput);
+                addTask(TaskType.EVENT, taskToAdd, list);
+            } else {
+                System.out.println(HORIZONTAL_LINE + HELP_MESSAGE + HORIZONTAL_LINE);
+            }
+        }
+    }
+
+    /**
+     * Adds a task to an array list of tasks, printing out the newly added to task to the terminal
+     *
+     * @param type        The type of task (using the TaskType enumerator)
+     * @param description Description of the task to be added
+     * @param taskList    Array list of tasks
+     */
+    private static void addTask(TaskType type, String description, List<Task> taskList) {
```

Comment:
maybe can refactor helper functions into a new class
Ex: helper function for tasks can be put in class TaskManager